packages/entities/src/lib/update.mutation.ts

```diff
@@ -243,3 +243,85 @@ export function upsertEntities<
     };
   };
 }
+
+/**
+ * Update entities ids
+ *
+ * @example
+ *
+ * // update single entity id
+ * store.update(updateEntitiesIds(1, 2));
+ *
+ * // update multiple entities ids
+ * store.update(updateEntitiesIds([1, 2], [10, 20]));
+ *
+ * // update entity id using a custom ref
+ * store.update(updateEntitiesIds(1, 2, { ref: UIEntitiesRef }));
+ *
+ */
+export function updateEntitiesIds<
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  oldId: OrArray<getIdType<S, Ref>>,
+  newId: OrArray<getIdType<S, Ref>>,
+  options: BaseEntityOptions<Ref> = {}
+): Reducer<S> {
+  return function (state, context) {
+    const oldIds = coerceArray(oldId);
+    const newIds = coerceArray(newId);
+
+    if (oldIds.length !== newIds.length) {
+      throw new Error('The number of old and new ids must be equal');
+    }
+
+    const { ref = defaultEntitiesRef } = options;
+    const idProp = getIdKey<string>(context, ref);
+    const updatedEntities = { ...state[ref.entitiesKey] };
+
+    for (let i = 0; i < oldIds.length; i++) {
+      const oldVal = oldIds[i];
+      const newVal = newIds[i];
+
+      if (state[ref.entitiesKey][newVal]) {
```

Comment:
Let's remove this protection, it's redundant.

Reply:
This scenario is hopefully rare, but if the code does not throw, it could lead to some nasty results.
Imagine a scenario:
```ts
// the user wants to update id 2 to become 3 and then id 1 to become 2
// the following would work
store.update(
  updateEntitiesIds(2, 3),
  updateEntitiesIds(1, 2)
);
```
however, if they have code like below...
```ts
store.update(updateEntitiesIds([1, 2], [2, 3]));
```
... the operation would succeed and this would be the store value:

```ts
{
  entities: { '3': { id: 3, title: 'todo 2', completed: false } },
  ids: [ 2, 3 ]
}
```
As you can see, the store is out of control.
I, as a developer, would be happy to see this error and figure out why my code is trying to update to ids that already exist.